bats/keyless-tables.bats

```diff
@@ -0,0 +1,369 @@
+    dolt commit -am "inserted on other"
+
+    run dolt merge master
+    [ $status -eq 0 ]
+    run dolt sql -q "SELECT * FROM keyless WHERE c0 > 6;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "7,7" ]] || false
+    [[ "$output" = "9,9" ]] || false
+    [[ "$output" = "8,8" ]] || false
+    [[ "$output" = "9,9" ]] || false
+    [[ "$output" = "7,7" ]] || false
+}
+
+@test "keyless diff branches with convergent mutation history" {
+    dolt branch other
+
+    dolt sql -q "INSERT INTO keyless VALUES (7,7),(8,8),(9,9);"
+    dolt commit -am "inserted on master"
+
+    dolt checkout other
+    dolt sql <<SQL
+INSERT INTO keyless VALUES (9,9),(8,8),(7,7),;
+UPDATE keyless SET (c0,c1) = (7,7) WHERE c0 = 9;
+UPDATE keyless SET (c0,c1) = (9,9) WHERE c0 = 7;
+SQL
+    dolt commit -am "inserted on other"
+
+    run dolt diff master
+    [ $status -eq 0 ]
+    [ $output = "" ]
+}
+
+@test "keyless merge branches with convergent mutation history" {
+    dolt branch other
+
+    dolt sql -q "INSERT INTO keyless VALUES (7,7),(8,8),(9,9);"
+    dolt commit -am "inserted on master"
+
+    dolt checkout other
+    dolt sql <<SQL
+INSERT INTO keyless VALUES (9,9),(8,8),(7,7),;
+UPDATE keyless SET (c0,c1) = (7,7) WHERE c0 = 9;
+UPDATE keyless SET (c0,c1) = (9,9) WHERE c0 = 7;
+SQL
+    dolt commit -am "inserted on other"
+
+    run dolt merge master
+    [ $status -eq 0 ]
+    run dolt sql -q "SELECT * FROM keyless WHERE c0 > 6;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "7,7" ]] || false
+    [[ "$output" = "8,8" ]] || false
+    [[ "$output" = "9,9" ]] || false
+}
+
+@test "keyless diff branches with offset mutation history" {
+    dolt branch other
+
+    dolt sql -q "INSERT INTO keyless VALUES (7,7),(8,8),(9,9);"
+    dolt commit -am "inserted on master"
+
+    dolt checkout other
+    dolt sql -q "INSERT INTO keyless VALUES (7,7),(7,7),(8,8),(9,9);"
+    dolt commit -am "inserted on other"
+
+    run dolt diff master
+    [ $status -eq 0 ]
+    [[ "$output" = "|  +  | 7  | 7  |" ]] || false
+    [[ "$output" = "|  -  | 8  | 8  |" ]] || false
+    [[ "$output" = "|  +  | 8  | 8  |" ]] || false
+    [[ "$output" = "|  -  | 9  | 9  |" ]] || false
+    [[ "$output" = "|  +  | 9  | 9  |" ]] || false
+}
+
+
+@test "keyless merge branches with offset mutation history" {
+    dolt branch other
+
+    dolt sql -q "INSERT INTO keyless VALUES (7,7),(8,8),(9,9);"
+    dolt commit -am "inserted on master"
+
+    dolt checkout other
+    dolt sql -q "INSERT INTO keyless VALUES (7,7),(7,7),(8,8),(9,9);"
+    dolt commit -am "inserted on other"
+
+    run dolt merge master
+    [ $status -eq 0 ]
+    run dolt sql -q "SELECT * FROM keyless WHERE c0 > 6;" -r csv
+    [ $status -eq 0 ]
+    [[ "$output" = "7,7" ]] || false
+    [[ "$output" = "7,7" ]] || false
+    [[ "$output" = "8,8" ]] || false
+    [[ "$output" = "8,8" ]] || false
+    [[ "$output" = "9,9" ]] || false
+    [[ "$output" = "9,9" ]] || false
+}
```

Comment:
Aaron is worried about deletes. I think we need delete tests. and more mixes of updates, inserts and deletes.

And I think regexes are going to be a problem specifically on tests where you assume order. I would use lines and equalities instead...at least lines.